kotlin-mbedtls-netty/src/test/kotlin/org/opencoap/ssl/netty/NettyTest.kt

```diff
@@ -224,4 +228,57 @@ class NettyTest {
         assertEquals(0, dtlsServer.numberOfSessions)
         client.close()
     }
+
+    @Test
+    fun `server should store session if hinted to do so`() {
+        val hintedSrvChannel = createBootstrap(
+            0,
+            DtlsChannelHandler(serverConf, sessionStore = sessionStore),
+            {
+                addLast(
+                    "echo",
+                    object : ChannelInboundHandlerAdapter() {
+                        override fun channelRead(ctx: ChannelHandlerContext, msg: Any) {
+                            val dgram = msg as DatagramPacket
+                            val dgramContent = dgram.content().toByteArray()
+                            val dgramSender = dgram.sender()
+                            val reply = ctx.alloc().buffer(dgram.content().readableBytes()).writeBytes(dgramContent)
+                            val goToSleep = dgramContent.toString(Charset.defaultCharset()).endsWith(":sleep")
+
+                            ctx.writeAndFlush(
+                                DatagramPacketWithContext(
+                                    reply,
+                                    dgramSender,
+                                    null,
+                                    DtlsSessionContext(sessionExpirationHint = goToSleep)
+                                )
+                            )
+                        }
+                    }
+                )
+            }
+        ).bind().sync().channel() as DatagramChannel
```

Comment:
What if you simply update `EchoHandler` with this logic?
Alternatively create an outbound handler (`ChannelOutboundHandler`) that would check for special string when writing, then add it in `beforeAll` method for `srvChannel`